src/types/rds_config.go

```diff
@@ -24,24 +32,38 @@ func (d *RdsConfig) IsEmpty() bool {
 		d.InstanceClass == "" &&
 		d.DbName == "" &&
 		d.Username == "" &&
-		d.PasswordEnvVar == "" &&
-		d.StorageType == ""
+		d.PasswordSecretName == "" &&
+		d.StorageType == "" &&
+		d.ServiceEnvVarNames.DbName == "" &&
+		d.ServiceEnvVarNames.Username == "" &&
+		d.ServiceEnvVarNames.Password == ""
 }
 
 // ValidateFields validates that an rds config contains all required fields
 func (d *RdsConfig) ValidateFields() error {
-	requiredFields := []string{"AllocatedStorage", "InstanceClass", "DbName", "Username", "PasswordEnvVar", "StorageType"}
+	requiredFields := []string{"AllocatedStorage", "InstanceClass", "DbName", "Username", "PasswordSecretName", "StorageType", "ServiceEnvVarNames"}
+	missingField := checkRequiredFields(*d, requiredFields)
+	if missingField != "" {
+		return fmt.Errorf("missing required field 'Rds.%s'", missingField)
+	}
+	dbNameRegex := regexp.MustCompile("^[a-zA-Z0-9-]+$")
+	if !dbNameRegex.MatchString(d.DbName) {
+		return errors.New("only alphanumeric characters and hyphens allowed in rds.db-name")
+	}
+	requiredServiceEnvVars := []string{"DbName", "Username", "Password"}
+	missingField = checkRequiredFields(d.ServiceEnvVarNames, requiredServiceEnvVars)
+	if missingField != "" {
+		return fmt.Errorf("missing required field 'Rds.ServiceEnvVarNames.%s", missingField) //TODO address camelcase issue
```

Comment:
Wanted to bring up that our error messages are in the form `Rds.ServiceEnvVarNames.DbName` when in a yaml folder they should be `rds.service-env-var-names.db-name`. Not sure if these are potentially misleading to the user.

Reply:
We should update them to the how they appear in the yaml file.